addons/project/models/project_task.py

```diff
@@ -1583,6 +1589,27 @@ def action_recurring_tasks(self):
             'domain': [('recurrence_id', 'in', self.recurrence_id.ids)],
         }
 
+    def action_project_sharing_recurring_tasks(self):
+        self.ensure_one()
+        recurrent_tasks = self.sudo().recurrence_id.task_ids
+        # if all the recurrent tasks are in the same project, open the view in edit mode.
+        if recurrent_tasks.project_id == self.project_id:
+            action = self.env['ir.actions.act_window']._for_xml_id('project.project_sharing_project_task_recurring_tasks_action')
+            action['context'] = {'default_project_id': self.project_id.id}
+            if len(recurrent_tasks) == 1:
+                action['view_mode'] = 'form'
+                action['views'] = [(view_id, view_type) for view_id, view_type in action['views'] if view_type == 'form']
+                action['res_id'] = recurrent_tasks.id
+            else:
+                action['domain'] = [('project_id', '=', self.project_id.id), ('display_in_project', '=', True), ('recurrence_id', 'in', self.recurrence_id.ids)]
```

Comment:
```suggestion
                action['domain'] = [
                    ('project_id', '=', self.project_id.id),
                    ('display_in_project', '=', True),
                    ('recurrence_id', 'in', self.recurrence_id.ids),
                ]
```
I think is more readable

Reply:
I think it is necessary when a complex expression is used. (or/else, etc). At least, from what i remember when i' added this part, and it seems to be similar in other part of the code (ex https://github.com/odoo/enterprise/blob/40e51c4a8fcf46815c9b5cbeaedbfb782dd510eb/helpdesk/data/mail_template_data.xml#L7C19-L7C19 )